ouroboros-consensus/src/Ouroboros/Consensus/Node/DbMarker.hs

```diff
@@ -141,6 +150,32 @@ instance Exception DbMarkerError where
       "Missing \"" <> f <> "\" but the folder was not empty"
     CorruptDbMarker f ->
       "Corrupt or unreadable \"" <> f <> "\""
+    FileLocked f ->
+      "The db is used by another process. File \"" <> f <> "\" is locked"
+
+{-------------------------------------------------------------------------------
+  Locking
+-------------------------------------------------------------------------------}
+
+-- | If the file is locked, it throws an exception.
+-- The file is unlocked when the @registry@ is closed.
+lockDbMarkerFile :: ResourceRegistry IO -> FilePath -> IO ()
+lockDbMarkerFile registry dbPath = do
+    mlockFile <- allocateEither
+                    registry
+                    (const $ justToRight <$> tryLockFile fullPath Exclusive)
+                    unlockFile
+    case mlockFile of
+      Left () -> throwIO $ FileLocked fullPath
+      Right _ -> return ()
+
+  where
+
```

Comment:
Remove the blank lines around `where`. If this function were large, then they would be more welcome.